src/providers/sendgrid.rs

```diff
@@ -82,3 +104,11 @@ impl From<SendgridError> for ProviderError {
         }
     }
 }
+
+impl From<Utf8Error> for ProviderError {
+    fn from(error: Utf8Error) -> ProviderError {
+        ProviderError {
+            description: format!("Failed to decode Sendgrid message id as UTF-8: {:?}", error),
```

Comment:
The text of this error message runs the risk of accidentally becoming out of date if we ever try to parse some other utf8 string from the sendgrid API, but use the same `From::from` error handling.  Not really sure how to make that better, just calling it out.